spec/mongoid/criteria/queryable/selector_spec.rb

```diff
@@ -403,6 +403,32 @@ def localized?
             end
           end
         end
+
+        context "when providing a Mongoid::RawValue" do
+
+          context "and raw_value is a string" do
+
+            before do
+              selector.send(method, "key", Mongoid::RawValue("Foo"))
+            end
+
+            it "returns the raw_value" do
+              expect(selector["key"]).to eq("Foo")
+            end
+          end
+
+          context "and raw_value is a complex type" do
+            config_override :map_big_decimal_to_decimal128, true
+
+            before do
+              selector.send(method, "key", Mongoid::RawValue([{ foo: 1, "Bar" => [/baz/, BigDecimal('2'), 4..5] }, 3]))
+            end
+
+            it "returns the raw_value" do
+              expect(selector["key"]).to eq([{ foo: 1, "Bar" => [/baz/, BigDecimal('2'), 4..5] }, 3])
```

Comment:
Can you also add a test which uses a type that isn't handled by bson-ruby and that is not transformed by Mongoid for RawValue logic, e.g. either a Range or `Object.new` for example.

Reply:
Done.